app/dimensions.tsx

```diff
@@ -22,10 +24,11 @@ export interface IDimensionsContextProps {
 
 export const DimensionsContext = React.createContext<Partial<IDimensionsContextProps>>(Dimensions.get('window'));
 
-export function withDimensions(Component: any): any {
-	const DimensionsComponent = (props: any) => (
+export function withDimensions<T extends object>(Component: React.ComponentType<T> & TNavigationOptions): typeof Component {
+	const DimensionsComponent = (props: T) => (
 		<DimensionsContext.Consumer>{contexts => <Component {...props} {...contexts} />}</DimensionsContext.Consumer>
 	);
+	DimensionsComponent.navigationOptions = Component.navigationOptions;
```

Comment:
We don't need this.
`hoitNonReactStatics`' goal is to do it.
https://reactjs.org/docs/higher-order-components.html#static-methods-must-be-copied-over

I commented it and it worked just fine.